src/components/feedback/Popover.tsx

```diff
@@ -0,0 +1,260 @@
+import classnames from 'classnames';
+import type { ComponentChildren, RefObject } from 'preact';
+import { useCallback, useLayoutEffect, useRef } from 'preact/hooks';
+
+import { ListenerCollection } from '../../util/listener-collection';
+
+/** Small space to apply between the anchor element and the popover */
+const POPOVER_ANCHOR_EL_GAP = '.15rem';
+
+/**
+ * Space in pixels to apply between the popover and the viewport sides to
+ * prevent it from growing to the very edges.
+ */
+export const POPOVER_VIEWPORT_HORIZONTAL_GAP = 8;
+
+type PopoverCSSProps =
+  | 'top'
+  | 'left'
+  | 'minWidth'
+  | 'marginBottom'
+  | 'bottom'
+  | 'marginTop';
+
+/**
+ * Manages the popover position manually to make sure it renders "next" to the
+ * anchor element (above or below). This is mainly needed when using the
+ * popover API, as that makes it render in the top layer, making it impossible
+ * to position it relative to the anchor element via regular CSS.
+ *
+ * @param asNativePopover - Native popover API is used to toggle the popover
+ * @param alignToRight - Whether the popover should be aligned to the right side
+ *                       of the anchor element or not
+ */
+function usePopoverPositioning(
+  anchorElementRef: RefObject<HTMLElement | undefined>,
+  popoverRef: RefObject<HTMLElement | undefined>,
+  popoverOpen: boolean,
+  asNativePopover: boolean,
+  alignToRight: boolean,
+) {
+  const adjustPopoverPositioning = useCallback(() => {
+    const popoverEl = popoverRef.current!;
+    const anchorEl = anchorElementRef.current!;
+
+    /**
+     * Set the positioning styles synchronously (not via `style` prop and a
+     * piece of state), to make sure positioning happens before other side
```

Comment:
This comment is slightly confusing on first read because the function does set the `style` JS property of the element. I understand that it means "set styles synchronously as opposed to via the `style` prop on the next render" though.

Reply:
True!

I did in fact mean to reference the `style` component prop (as in `Popover style={{ ... }} />`), as you mentioned, but this function does set a `style` property in an object, so it's misleading 😅 

Let me give it a thought.

Reply:
I changed the comment to `Set the positioning styles synchronously (not via <div style={computedStyles} />), to make sure positioning happens before other side effects.`